src/wallet/rpcwallet.cpp

```diff
@@ -3613,17 +3615,34 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         nTotalOut += nAmount;
     }
 
-    // Check the number of zaddr outputs does not exceed the limit.
-    if (zaddrRecipients.size() > Z_SENDMANY_MAX_ZADDR_OUTPUTS)  {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, too many zaddr outputs");
+    int nextBlockHeight = chainActive.Height() + 1;
+    CMutableTransaction mtx;
+    mtx.fOverwintered = true;
+    mtx.nVersionGroupId = SAPLING_VERSION_GROUP_ID;
```

Comment:
This determining of the version does not seem related to the transaction size limit.
Also wondering if it can be refactored to a generic method that does it according to the active upgrades

Reply:
Yeah, `CreateNewContextualCMutableTransaction()` would likely end up being what we want to use. There are more places than this PR though, so we should do so in a separate PR to ensure that it makes sense everywhere.